### PR TITLE
Refactor, simplify, remove Point, add neighbours

### PR DESCRIPTION
**Pull request by colega**
I managed to merge my branch rewriting changes instead of doing a plain merge.

What I did:
I changed the projection to Earth-size map to an [0,1]x[0,1] map, which made me remove all those multiplications by Ecuator length and Earth radius everywhere. After that, `Point` struct didn't make sense to be exported for end-users, so I shortened that step creating `Position`s from `LL`, and vice-versa.

With this changes, we don't need the `zooms` anymore, we just need a plain int describing the size of each level. We could actually use the `pow3` cache but I wanted to keep some semantic meaning in the places we're using it (and we're using it for quickly validating level)

Added `Neighbours` function that returns neighbours of an Hexagon. It won't work correctly at the poles and I don't want to invest time on that: using this library on the poles will be wrong anyways as they are very distorted with the projection being used.

Updated `README.md` with examples. Sorry for changing the format a little bit, but it's easier to copy-paste `example_test.go` with every change and fogetting about having to format it every time we change that one.

Added `doc.go` for small description at godoc.org.

Fixed Makefile adding `testdeps` recipe which is mentioned in README.md. 

Finally, I've added an [explanation of how encoding is done](https://github.com/cabify/go-geohex/tree/simplify-and-add-neighbours), as I didn't find it anywhere on the internet.

### CODE REVIEW

**Comment by colega** [comment]
Maybe this needs a version bump?

File: v3/example_test.go [hunk 1]
```diff
@@ -20,5 +20,20 @@ func ExampleDecode() {
 	fmt.Println(ll.Lat, ll.Lon)
 
 	// Output:
-	// 35.63992106908978 139.7256515775034
+	// 35.63992106908978 139.72565157750344
+}
+
+func ExampleNeighbours() {
```
Comment (dim) [inline v3/example_test.go]: according to godoc conventions, this needs to be `func ExamplePosition_Neighbours()`

File: README.md [hunk 2]
```diff
@@ -37,9 +65,63 @@ To run benchmarks, call:
...
+[Mercator projection](https://en.wikipedia.org/wiki/Mercator_projection), note
+that this means that hexagons at different latitudes cover different Earth areas.
+The coordinates on this square are called `e`, `n` in the code (east, north).
+
+Then we transform those coordinates into `x`, `y` coordinates on the hexagons map:
+we turn the axes 45 degrees and we stretch one of them by a factor of tan(pi/6),
+this makes each four adjacent hexagons centers' be equidistant. Then we decide
+which hexagon contains the desired coordinates. This is probably the trickiest part,
+we use this condition to decide:
+```go
```
Comment (dim) [inline README.md]: minor: I think code blocks needs spaces above and below to render in markdown correctly

File: v3/position_test.go [hunk 3]
```diff
@@ -27,3 +22,34 @@ var _ = Describe("Position", func() {
 	})
 
 })
+
+var _ = DescribeTable("Should return neighbors of Positions with correct coordinates",
```
Comment (dim) [inline v3/position_test.go]: minor: could you pls just move that within the `Describe("Position", ... ` block?

File: README.md (hunk 2 above)
Comment (colega) [inline README.md]: actually GitHub rendered this correctly, but I've added the spaces for better readability in plain text version.

File: v3/position_test.go (hunk 3 above)
Comment (colega) [inline v3/position_test.go]: Moved, and removed `var _ =` part as it's not needed there.

File: v3/example_test.go [hunk 4]
```diff
@@ -23,7 +23,7 @@ func ExampleDecode() {
 	// 35.63992106908978 139.72565157750344
 }
 
-func ExampleNeighbours() {
+func Example_Neighbours() {
```
Comment (dim) [inline v3/example_test.go]: sorry, still not right, https://blog.golang.org/examples#TOC_4.

**Comment by colega** [comment]
Fixed the benchmarks of `Decode` by adding back the premature position instantiation and operation over it.

File: v3/example_test.go (hunk 4 above)
Comment (colega) [inline v3/example_test.go]: omg... sorry :(

**Comment by dim** [comment]
k, I yanked the other release and will release the ^ on Friday (will give us a few days to reflect)

**Comment by dim** [comment]
Pls feel free to submit a PR for a Contributors section in the README :)

**Pull request merged**